test/e2e/e2e_visibility_on_demand_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	//"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/kueue-operator/test/e2e/testutils"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	flowcontrolclientv1 "k8s.io/client-go/kubernetes/typed/flowcontrol/v1"
+)
+
+const (
+	priorityName = "kueue-visibility"
+)
+
+var _ = Describe("VisibilityOnDemand", Label("visibility-on-demand"), Ordered, func() {
+	BeforeAll(func() {
+		Expect(deployOperand()).To(Succeed(), "operand deployment should not fail")
+	})
+	AfterAll(func(ctx context.Context) {
+		testutils.CleanUpKueueInstance(ctx, clients.KueueClient, "cluster")
+	})
+
+	When("kueue.openshift.io/allow-nominal-concurrency-shares-update annotation is set to true", func() {
+		labelKey := "kueue.openshift.io/managed"
+		labelValue := "true"
+		testQueue := "test-queue"
+
+		var cleanupClusterQueue, cleanupLocalQueue, cleanupResourceFlavor func()
+		var err error
+		var ns *corev1.Namespace
+		var priorityClient flowcontrolclientv1.PriorityLevelConfigurationInterface
+
+		BeforeAll(func(ctx context.Context) {
+			priorityClient = clients.KubeClient.FlowcontrolV1().PriorityLevelConfigurations()
+			ns, err = kubeClient.CoreV1().Namespaces().Create(context.TODO(), &corev1.Namespace{
```

Comment:
Should we have a create and wait for object to exist?

I don't think create is a blocking call.